fairlearn/metrics/_metrics_engine.py

```diff
@@ -3,104 +3,124 @@
 
 import numpy as np
 
-from ._group_metric_result import GroupMetricResult
 from ._input_manipulations import _convert_to_ndarray_and_squeeze
+from sklearn.utils import Bunch
 
 _MESSAGE_SIZE_MISMATCH = "Array {0} is not the same size as {1}"
 
+# Parameters to metrics that should be split according to sensitive features
+_SPLITTABLE_METRIC_PARAMS = {"sample_weight"}
```

Comment:
Isn't this something which should be specified when the metric is invoked?

Reply:
Do you mean when the metric is invoked or when the group metric function is created?

What I have currently functionally matches our previous implementation. I think a better solution would be to allow providing alternative lists when calling `make_group_metric`. But I'd leave that for a future PR.

Reply:
Would 'future' be 'immediately after this?' I can see the sense in doing this in two stages (less chance for disaster), but I don't think that it should be delayed for long.

I think that this would be specified when the group metric function is created, now you point out the distinction. After all, at that point the underlying metric is bound, and its arguments aren't going to change.

Reply:
yes... that's what i was thinking. i think it's easy enough to implement, but i might leave proper testing to a future PR.